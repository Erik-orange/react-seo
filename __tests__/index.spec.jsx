/*
 * Copyright 2020 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import SEO from '../src';

jest.mock('react-helmet', () => ({ Helmet: 'Helmet' }));

describe('SEO', () => {
  it('should render the snapshot correctly', () => {
    const component = shallow(
      <SEO
        author="John Doe"
        description="Lorem ipsum sat delor."
        keywords={['foo', 'bar']}
        siteUrl="https://example.com"
        title="Lorem Ipsum"
        canonical="https://example.com/foo/bar"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render articles correctly', () => {
    const component = shallow(
      <SEO
        author="John Doe"
        description="Lorem ipsum sat delor."
        keywords={['foo', 'bar']}
        siteUrl="https://example.com"
        title="Lorem Ipsum"
        article={true}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render images correctly', () => {
    const component = shallow(
      <SEO
        author="John Doe"
        description="Lorem ipsum sat delor."
        keywords={['foo', 'bar']}
        siteUrl="https://example.com"
        title="Lorem Ipsum"
        image={{
          src: '/foo.png',
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render images pathnames correctly', () => {
    const component = shallow(
      <SEO
        author="John Doe"
        description="Lorem ipsum sat delor."
        keywords={['foo', 'bar']}
        siteUrl="https://example.com"
        title="Lorem Ipsum"
        pathname="/foo"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const component = shallow(
      <SEO
        author="John Doe"
        description="Lorem ipsum sat delor."
        keywords={['foo', 'bar']}
        siteUrl="https://example.com"
        title="Lorem Ipsum"
        pathname="/foo"
      >
        <script src="https://example.com/foo.js" />
      </SEO>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render Helmet props correctly', () => {
    const component = shallow(
      <SEO
        author="John Doe"
        description="Lorem ipsum sat delor."
        keywords={['foo', 'bar']}
        siteUrl="https://example.com"
        title="Lorem Ipsum"
        canonical="https://example.com/foo/bar"
        defaultTitle="Curabitur pretium tincidunt lacus."
        defer={false}
        encodeSpecialCharacters={true}
        onChangeClientState={jest.fn()}
        titleTemplate={null}
        zipTies={false}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
