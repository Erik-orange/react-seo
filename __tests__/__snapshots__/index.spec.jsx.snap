// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO should render articles correctly 1`] = `
<Helmet
  htmlAttributes={
    Object {
      "lang": "en-US",
    }
  }
  link={Array []}
  meta={
    Array [
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "description",
      },
      Object {
        "content": "foo,bar",
        "name": "keywords",
      },
      Object {
        "content": "Lorem Ipsum",
        "property": "og:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "property": "og:description",
      },
      Object {
        "content": "article",
        "property": "og:type",
      },
      Object {
        "content": "John Doe",
        "name": "twitter:creator",
      },
      Object {
        "content": "Lorem Ipsum",
        "name": "twitter:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "twitter:description",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
    ]
  }
/>
`;

exports[`SEO should render children correctly 1`] = `
<Helmet
  htmlAttributes={
    Object {
      "lang": "en-US",
    }
  }
  link={
    Array [
      Object {
        "href": "https://example.com/foo",
        "rel": "canonical",
      },
    ]
  }
  meta={
    Array [
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "description",
      },
      Object {
        "content": "foo,bar",
        "name": "keywords",
      },
      Object {
        "content": "Lorem Ipsum",
        "property": "og:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "John Doe",
        "name": "twitter:creator",
      },
      Object {
        "content": "Lorem Ipsum",
        "name": "twitter:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "twitter:description",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
    ]
  }
>
  <script
    src="https://example.com/foo.js"
  />
</Helmet>
`;

exports[`SEO should render images correctly 1`] = `
<Helmet
  htmlAttributes={
    Object {
      "lang": "en-US",
    }
  }
  link={Array []}
  meta={
    Array [
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "description",
      },
      Object {
        "content": "foo,bar",
        "name": "keywords",
      },
      Object {
        "content": "Lorem Ipsum",
        "property": "og:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "John Doe",
        "name": "twitter:creator",
      },
      Object {
        "content": "Lorem Ipsum",
        "name": "twitter:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "twitter:description",
      },
      Object {
        "content": "https://example.com/foo.png",
        "property": "og:image",
      },
      Object {
        "content": undefined,
        "property": "og:image:width",
      },
      Object {
        "content": undefined,
        "property": "og:image:height",
      },
      Object {
        "content": "summary_large_image",
        "name": "twitter:card",
      },
    ]
  }
/>
`;

exports[`SEO should render images pathnames correctly 1`] = `
<Helmet
  htmlAttributes={
    Object {
      "lang": "en-US",
    }
  }
  link={
    Array [
      Object {
        "href": "https://example.com/foo",
        "rel": "canonical",
      },
    ]
  }
  meta={
    Array [
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "description",
      },
      Object {
        "content": "foo,bar",
        "name": "keywords",
      },
      Object {
        "content": "Lorem Ipsum",
        "property": "og:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "John Doe",
        "name": "twitter:creator",
      },
      Object {
        "content": "Lorem Ipsum",
        "name": "twitter:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "twitter:description",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
    ]
  }
/>
`;

exports[`SEO should render the snapshot correctly 1`] = `
<Helmet
  htmlAttributes={
    Object {
      "lang": "en-US",
    }
  }
  link={
    Array [
      Object {
        "href": "https://example.com/foo/bar",
        "rel": "canonical",
      },
    ]
  }
  meta={
    Array [
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "description",
      },
      Object {
        "content": "foo,bar",
        "name": "keywords",
      },
      Object {
        "content": "Lorem Ipsum",
        "property": "og:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "property": "og:description",
      },
      Object {
        "content": "website",
        "property": "og:type",
      },
      Object {
        "content": "John Doe",
        "name": "twitter:creator",
      },
      Object {
        "content": "Lorem Ipsum",
        "name": "twitter:title",
      },
      Object {
        "content": "Lorem ipsum sat delor.",
        "name": "twitter:description",
      },
      Object {
        "content": "summary",
        "name": "twitter:card",
      },
    ]
  }
/>
`;
